Adds the ability to override Locations that have items that are always available

misilot opened the pull request:
Adds the ability to override Locations that have items that are always available, but not for loan, so they show up again as available, and not checked out. 

This is a limitation of how Koha / VuFind handles items that are not for loan but are actually there.

If the item is lost/missing it is still accounted for.

demiankatz commented:
Thanks for this! A few comments:

1.) I notice that in the code you are replacing, ONLINE and INTERNET are hard-coded as available locations. However, your new configuration uses REFERENCEAREA as a default setting. Should we be putting ONLINE / INTERNET in the .ini file instead/in addition? Should we default to ONLINE and INTERNET as the available locations if the .ini setting is commented out?

2.) The comment about ONLINE and INTERNET should be updated.

3.) There's a typo "not not" in one of the comments.

4.) Looks like some style fixes are needed; please let me know if I can be of any assistance.

misilot commented:
1. ONLINE and INTERNET  were our location names, they can be anything. I was trying to put a real life in the library use case, usually reference collections are "not for loan" but are available. I can put ONLINE and INTERNET if that makes more sense, I can change it.
2. Fixed
3. Fixed
4. Currently checking this.

demiankatz commented:
Maybe the thing to do is put all three examples, ONLINE, INTERNET and REFERENCEAREA into the ..ini file, but all commented out by default. That way we can emphasize the fact that users need to fill this in with their own location codes, but these examples show the types of things that might need this functionality.

misilot commented:
Done, waiting for Travis to run

demiankatz commented:
Thanks. Looks like there's a syntax error here -- missing close parenthesis on line 749 (and the brace on 750 should be up on the end of 749 if it fits).

misilot commented:
Moved the brace and added closing ). Waiting on jenkins again.

demiankatz commented:
Now phpcs is happy, but it looks like it needs a run through php-cs-fixer. If you're not set up to do that yourself, let me know and I'm happy to take care of it as part of the merging process.

misilot commented:
Should be good now?

demiankatz commented:
Looks good to me!

misilot commented:
Great thanks!
